CategoryMicro/src/main/java/com/grizzly/categorymicro/CategoryService.java

```diff
@@ -103,75 +119,67 @@ public void deleteById(Integer deleteId) {
      * @param id, the string to match to ID to filter the product by
      * @return Category
      */
-    public Category getSingle(Integer id) {
-        return categoryRepository.findById(id).get();
+    public CategoryDTO getSingle(Integer id) {
+        return categoryToDTO(categoryRepository.findById(id).get());
     }
 
     /**
      * Get a filtered list of categories, based on a given search string to match to name .
      * @param search, the string to match to name to filter the categories by
      * @return ArrayList of Category objs whose names match
      */
-    public ArrayList<Category> getFiltered(String search) {
+    public ArrayList<CategoryDTO> getFiltered(String search) {
         try {
             Integer categoryId = Integer.parseInt(search);
 
             // To meet the specifications of spitting out an ArrayList...
-            ArrayList<Category> category = new ArrayList<>();
+            ArrayList<CategoryDTO> category = new ArrayList<>();
             category.add(getSingle(categoryId));
 
             return category;
         } catch(NumberFormatException e) {
             Sort sort = new Sort(Sort.Direction.ASC, "categoryId");
             PageRequest request = PageRequest.of(0, 25, sort);
 
-            return makeListFromIterable(
-                    categoryRepository.findByCategoryName(search, request)
-            );
+            List<Category> categories = categoryRepository.findByCategoryName(search, request);
+
+            ArrayList<CategoryDTO> categoryResult = new ArrayList<>();
+            for (Category category : categories) {
+                categoryResult.add(categoryToDTO(category));
+            }
+
+            return categoryResult;
         }
     }
     /**
      * Get a list of vendors based on vendor IDs
      * @param ids, The list of Vendor ids that are to be fetched
      * @return ArrayList of Vendor objs whose IDs match ids
      */
-    public ArrayList<Category> getBatchbyId(List<String> ids) {
-        List<Integer> ints = new ArrayList<Integer>();
+    public ArrayList<CategoryDTO> getBatchbyId(List<String> ids) {
+        List<Integer> categoryIds = new ArrayList<>();
 
-        for(String id: ids){
-            ints.add(Integer.parseInt(id));
+        for (String id: ids){
+            categoryIds.add(Integer.parseInt(id));
         }
-        ArrayList<Category> batchList = makeListFromIterable(categoryRepository.findByCategoryIdIn(ints));
```

Comment:
Why has this data validation been deleted?

Reply:
because it has been changed to DTO. github changelog sucks, it's not showing my previous commits changes.

Reply:
view full file for actual content

Reply:
I mean the validation for invalid IDs passed in. With your committed file, if I pass in '-1,3,5' I will get back 2 items, but I will be expected 3. This could cause serious issues. With the validation that was there, for -1 we were adding a 'blank' item which wouldn't mess anything up on the front end, it would just show up as a category not found in the category column.